### PR TITLE
Blob write optimisation

### PR DESCRIPTION
**Pull request by mugurrus**

### CODE REVIEW

**Comment by mgax** [comment]
Wait, did we not merge this already in https://github.com/hoover/snoop2/pull/137?

File: snoop/data/models.py [hunk 1]
```diff
@@ -138,21 +138,24 @@ def create_from_bytes(cls, data):
     @classmethod
     def create_from_file(cls, path):
         file_sha3_256 = hashlib.sha3_256()
-        with open(path, 'rb') as f:
-            for block in chunks(f):
-                file_sha3_256.update(block)
 
         try:
-            return Blob.objects.get(pk=file_sha3_256.hexdigest())
+            blob = Blob.objects.get(pk=file_sha3_256.hexdigest())
```
Comment (gabriel-v) [inline snoop/data/models.py]: The hexdigest here is always uninitialized, as you moved the loop that updates it under this line.

File: snoop/data/models.py [hunk 2]
```diff
@@ -138,21 +138,24 @@ def create_from_bytes(cls, data):
...
+            with open(path, 'rb') as f:
+                for block in chunks(f):
+                    file_sha3_256.update(block)
 
         except ObjectDoesNotExist:
             with cls.create() as writer:
                 with open(path, 'rb') as f:
                     for block in chunks(f):
+                        file_sha3_256.update(block)
```
Comment (gabriel-v) [inline snoop/data/models.py]: The hash is not used anywhere in this code. The `writer` will calculate its own hashes. How is this needed?

**Comment by mgax** [comment]
Can we close this PR? It's been haunting us for two months now.

**Pull request closed without merging**